lib/cocoapods/installer/xcode/pods_project_generator/pod_target_integrator.rb

```diff
@@ -211,14 +212,57 @@ def add_prepare_artifacts_script_phase(native_target, spec)
+          #
+          # @param [PBXNativeTarget] native_target
+          #         the native target for which to add the the copy dSYM files build phase.
+          #
+          # @return [void]
+          #
+          def add_copy_strip_dsyms_script_phase(native_target)
+            dsym_paths = target.framework_paths.values.flatten.reject { |fmwk_path| fmwk_path.dsym_path.nil? }.map(&:dsym_path)
+
+            if dsym_paths.empty?
+              UserProjectIntegrator::TargetIntegrator.remove_copy_resources_script_phase_from_target(native_target)
+              script_phase = native_target.shell_script_build_phases.find { |bp| bp.name && bp.name.end_with?(UserProjectIntegrator::TargetIntegrator::COPY_DSYM_FILES_PHASE_NAME) }
+              native_target.build_phases.delete(script_phase) if script_phase.present?
+              return
+            end
+
+            phase_name = UserProjectIntegrator::TargetIntegrator::BUILD_PHASE_PREFIX + UserProjectIntegrator::TargetIntegrator::COPY_DSYM_FILES_PHASE_NAME
+            phase = UserProjectIntegrator::TargetIntegrator.create_or_update_shell_script_build_phase(native_target, phase_name)
+            phase.shell_script = dsym_script_contents(dsym_paths)
+
+            input_paths_by_config = {}
+            output_paths_by_config = {}
+            if use_input_output_paths?
+              input_file_list_path = target.strip_dsym_script_input_files_path
+              input_file_list_relative_path = "${PODS_ROOT}/#{input_file_list_path.relative_path_from(target.sandbox.root)}"
+              input_paths_key = UserProjectIntegrator::TargetIntegrator::XCFileListConfigKey.new(input_file_list_path, input_file_list_relative_path)
+              input_paths = input_paths_by_config[input_paths_key] = []
+              input_paths.concat dsym_paths
+
+              output_file_list_path = target.strip_dsym_script_input_files_path
+              output_file_list_relative_path = "${PODS_ROOT}/#{output_file_list_path.relative_path_from(target.sandbox.root)}"
+              output_paths_key = UserProjectIntegrator::TargetIntegrator::XCFileListConfigKey.new(output_file_list_path, output_file_list_relative_path)
+              output_paths = output_paths_by_config[output_paths_key] = []
+
+              dsym_output_paths = dsym_paths.map { |dsym_path| "${TARGET_BUILD_DIR}/#{File.basename(dsym_path)}" }
+              output_paths.concat dsym_output_paths
+            end
+
+            UserProjectIntegrator::TargetIntegrator.set_input_output_paths(phase, input_paths_by_config, output_paths_by_config)
+          end
+
           # @return [String] the message that should be displayed for the target
           #         integration.
           #
           def integration_message
             "Integrating target `#{target.name}`"
           end
 
-          # @return [PodTarget] the target part of the installation result.
+          # @return [Target] the target part of the installation result.
```

Comment:
is this correct? this _is_ `PodTargetIntegrator` hah

Reply:
yes this is correct I was fixing an IDE warning.